7a_temp_coop_munge/src/data_parsers/parse_navico_files.R

```diff
@@ -0,0 +1,36 @@
+parse_navico_files <- function(inind, outind){
+
+  infile <- sc_retrieve(inind, remake_file = '6_temp_coop_fetch_tasks.yml')
+  outfile <- as_data_file(outind)
+
+  # standardize column types - required because a subset of
+  # numeric columns contain text. Text in data columns will be coerced to `NA`
+  cols <- c(rep('text', 2), # state and wb name
+            rep('numeric', 18), # all data columns and some metadata
+            rep('date', 2)) # dates
+
+  # read in data: identify sheets in data, read in raw data by sheet, and combine into one data set
+  raw_sheets <- readxl::excel_sheets(infile)
+  ls_raw <- lapply(raw_sheets, function(x){
+    out <- read_xlsx(infile, sheet = x, col_types = cols)
+    return(out)
+  })
+  raw <- bind_rows(ls_raw)
```

Comment:
Since you specifically asked about style, I wanted to share that you can actually use a `%>%` to go from the output of `lapply` to `bind_rows()` and skip the intermediate object. And actually, you can go one step further and pipe the raw_sheets to the `lapply` to chunk all of the raw data steps into one thing. And inside of the `lapply` since it is only one line, writing an object `out` isn't necessary. We usually only do that if we have multiple lines and we need to clearly say what is coming back from the function or the function we called doesn't return the thing we need it to (as we do when we write out files and need the filepath returned).

```r
raw_data <- readxl::excel_sheets(infile) %>% 
    lapply(function(x){
        read_xlsx(infile, sheet = x, col_types = cols)
    }) %>%
    bind_rows()
```

Reply:
This is good to know. I wasn't aware that base R functions played well with pipelines. Thanks!

Reply:
Yep! The `%>%` is essentially just passing the output from the function on the left/above as input to the _first_ argument in the function on the right/below. So, it can work with in a lot of different scenarios!